scripts/variantstore/wdl/extract/gvs_avros_to_vds.py

```diff
@@ -313,35 +324,53 @@ def add_reference_allele(mt):
-        vd = vd.annotate_rows(
-            allele_NO=vd.alleles[1:].map(
-                lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'N', False)),
-            allele_YES=vd.alleles[1:].map(
-                lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'Y', True)),
-            allele_is_snp=is_snp,
-            allele_OK=hl._zip_func(is_snp, vd.alleles[1:],
-                                   f=lambda is_snp, alt:
-                                   hl.coalesce(vd.as_vqsr.get(alt).vqslod >=
-                                               hl.if_else(is_snp, vd.snp_vqslod_threshold, vd.indel_vqslod_threshold),
-                                               True))
-        )
+        if use_classic_vqsr:
+            vd = vd.annotate_globals(tranche_data=tranche.collect(_localize=False),
+                                     truth_sensitivity_snp_threshold=truth_sensitivity_snp_threshold,
+                                     truth_sensitivity_indel_threshold=truth_sensitivity_indel_threshold)
+
+
+            sorted_tranche_data = hl.sorted(vd.tranche_data, key=lambda x: x.truth_sensitivity)
+            vd = vd.annotate_globals(snp_vqslod_threshold=
+                                     sorted_tranche_data.filter(lambda x: (x.model == 'SNP') & (
+                                                 x.truth_sensitivity >= truth_sensitivity_snp_threshold))
+                                     .head().min_vqslod
+                                     ,
+                                     indel_vqslod_threshold=sorted_tranche_data.filter(lambda x: (x.model == 'INDEL') & (
+                                                 x.truth_sensitivity >= truth_sensitivity_indel_threshold))
+                                     .head().min_vqslod
+                                     )
+
+            is_snp = vd.alleles[1:].map(lambda alt: hl.is_snp(vd.alleles[0], alt))
+            vd = vd.annotate_rows(
+                allele_NO=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'N', False)),
+                allele_YES=vd.alleles[1:].map(
+                    lambda allele: hl.coalesce(vd.as_vqsr.get(allele).yng_status == 'Y', True)),
+                allele_is_snp=is_snp,
+                allele_OK=hl._zip_func(is_snp, vd.alleles[1:],
+                                       f=lambda is_snp, alt:
+                                       hl.coalesce(vd.as_vqsr.get(alt).vqslod >=
+                                                   hl.if_else(is_snp, vd.snp_vqslod_threshold, vd.indel_vqslod_threshold),
+                                                   True))
+            )
+        else:
+            vd = vd.annotate_globals(truth_sensitivity_snp_threshold=truth_sensitivity_snp_threshold,
+                                     truth_sensitivity_indel_threshold=truth_sensitivity_indel_threshold)
```

Comment:
I think it may be possible to DRY this wrt lines 328-300 as that block looks the same with the addition of the tranches.